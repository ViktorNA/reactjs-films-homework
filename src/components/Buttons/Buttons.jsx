import React from 'react';
import propTypes from 'prop-types';
import cn from 'classnames';
import styles from './buttons.scss';

const Button = (props) => {
  const {
    type,
    text,
    action,
    additionStyle,
  } = props;
  const classes = cn(styles[type], styles.common);
  return <button type="button" className={classes} onClick={action} style={additionStyle}>{text}</button>;
};

export default Button;

Button.propTypes = {
  type: propTypes.string.isRequired,
  text: propTypes.string,
  action: propTypes.func.isRequired,
  additionStyle: propTypes.PropTypes.objectOf(propTypes.string),
};

Button.defaultProps = {
  text: 'Watch Now',
  additionStyle: {},
};
