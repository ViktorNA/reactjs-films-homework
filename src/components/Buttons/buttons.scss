.common {
  cursor: pointer;
  font-size: 14px;
  color: white;
  border-radius: 3px;
  width: 100px;
  height: 30px;
}

.watchNow {
  border: 0;
  background-color: #3bb4e0;
}

.viewInfo {
  border: 1px solid white;
  background-color: rgba(255, 255, 255, 0);
  align-content: center;
  grid-column: 2/3;
}

.hidden {
  display: none;
}
