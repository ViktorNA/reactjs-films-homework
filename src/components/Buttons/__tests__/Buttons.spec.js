import TestRenderer from 'react-test-renderer';
import React from 'react';
import Buttons from '../index';

describe('Snapshot test', () => {
  it('watchNow works correctly', () => {
    const testRender = TestRenderer.create(<Buttons type="watchNow" />);
    expect(testRender.toJSON()).toMatchSnapshot();
  });
  it('viewInfo works correctly', () => {
    const testRender = TestRenderer.create(<Buttons type="viewInfo" />);
    expect(testRender.toJSON()).toMatchSnapshot();
  });
});
