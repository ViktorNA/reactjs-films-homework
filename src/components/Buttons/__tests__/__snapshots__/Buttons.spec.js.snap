// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test viewInfo works correctly 1`] = `
<button
  className="viewInfo common"
  style={Object {}}
  type="button"
>
  Watch Now
</button>
`;

exports[`Snapshot test watchNow works correctly 1`] = `
<button
  className="watchNow common"
  style={Object {}}
  type="button"
>
  Watch Now
</button>
`;
